int32 vp_x
int32 vp_y
float32 left_angle
float32 right_angle
bool vp_exists